src/main/java/edu/wpi/first/pathweaver/Path.java

```diff
@@ -1,126 +1,97 @@
+        new Point2D(0, 10), name);
   }
 
   /**
    * Path constructor based on known start and end points.
    *
-   * @param start        The starting waypoint of new path
-   * @param end          The ending waypoint of new path
+   * @param startPos        The starting waypoint of new path
+   * @param endPos          The ending waypoint of new path
    * @param startTangent The starting tangent vector of new path
    * @param endTangent   The ending tangent vector of new path
    * @param name         The string name to assign path, also used for naming exported files
    */
-  public Path(Point2D start, Point2D end, Point2D startTangent, Point2D endTangent, String name) {
+  public Path(Point2D startPos, Point2D endPos, Point2D startTangent, Point2D endTangent, String name) {
     pathName = name;
-    createInitialWaypoints(start, end, startTangent, endTangent);
+    Waypoint start = new Waypoint(startPos, startTangent, false, this);
+    Waypoint end = new Waypoint(endPos, endTangent, false, this);
+    start.setSpline(new QuickSpline(start, end));
+    waypoints.add(start);
+    waypoints.add(end);
+    updateSplines();
   }
 
   @Override
   public String toString() {
     return getPathName();
   }
 
-  private void createDefaultWaypoints() {
-    Point2D startPos = new Point2D(0, 0);
-    Point2D endPos = new Point2D(10, 10);
-
-    Point2D startTangent = new Point2D(10, 0);
-    Point2D endTangent = new Point2D(0, 10);
-    createInitialWaypoints(startPos, endPos, startTangent, endTangent);
-  }
-
-  private void createInitialWaypoints(Point2D startPos, Point2D endPos, Point2D startTangent, Point2D endTangent) {
-    start = new Waypoint(startPos, startTangent, false, this);
-    end = new Waypoint(endPos, endTangent, false, this);
-
-    start.setNextWaypoint(end);
-    end.setPreviousWaypoint(start);
-    createCurve(start, end);
-  }
-
   /**
-   * Make new Spline object.
+   * Creates new Waypoint in Path after previous.
    *
-   * @param start First Waypoint connected
-   * @param end   Second Waypoint connected
-   *
-   * @return Spline object
-   */
-  public Spline createCurve(Waypoint start, Waypoint end) {
-    Spline curve = new Spline(start, end);
-    curve.getCubic().toBack();
-    return curve;
-  }
-
-  /**
-   * Add Waypoint between previous and next.
-   *
-   * @param previous The point prior to new point
-   * @param next     The point after the new point
+   * @param previous The node before this one
+   * @param position Position to play new Waypoint
+   * @param tangent  Tangent vector at the new point
    *
    * @return new Waypoint
    */
-  public Waypoint addNewWaypoint(Waypoint previous, Waypoint next) {
-    if (previous.getNextWaypoint() != next || next.getPreviousWaypoint() != previous) {
-      throw new IllegalArgumentException("New Waypoint not between connected points");
+  public Waypoint addNewWaypoint(Waypoint previous, Point2D position, Point2D tangent, Boolean locked) {
```

Comment:
Use `boolean` primitive. `Boolean` should _only_ be used when working with generics where primitives are not supported